File: swift/Poly Canyon/Routing/DesignVillage/DVOnboarding.swift
import SwiftUI

struct DVOnboarding: View {
    @AppStorage("DVOnboardingComplete") var onboardingComplete: Bool = false
    @State private var currentSlide: Int = 0
    @Binding var userRole: DVRole
    
    var body: some View {
        ZStack {
            // Content slides
            if currentSlide == 0 {
                DVWelcomeSlide {
                    currentSlide = 1
                }
            } else if currentSlide == 1 {
                DVOverviewSlide(
                    onNext: { currentSlide = 2 },
                    onBack: { currentSlide = 0 }
                )
            } else if currentSlide == 2 {
                DVRulesSlide(
                    selectedRole: $userRole,
                    onNext: { currentSlide = 3 },
                    onBack: { currentSlide = 1 }
                )
            } else if currentSlide == 3 {
                DVPolyCanyonSlide(
                    onNext: { currentSlide = 4 },
                    onBack: { currentSlide = 2 }
                )
            } else if currentSlide == 4 {
                DVFinalSlide(onComplete: completeOnboarding)
            }
            
            // Progress indicator
            if currentSlide < 4 {
                VStack {
                    DVOnboardingIndicator(
                        totalStages: 5,
                        currentStage: currentSlide
                    )
                    .padding(.top, 20)
                    
                    Spacer()
                }
            }
        }
        .animation(.easeInOut, value: currentSlide)
        .transition(.opacity)
        .nexusStyle()
    }
    
    private func completeOnboarding() {
        withAnimation {
            onboardingComplete = true
        }
    }
}

enum DVRole: String {
    case visitor
    case competitor
}

struct DVWelcomeSlide: View {
    let onNext: () -> Void
    
    var body: some View {
        DVBaseSlide(
            buttonText: "Next",
            buttonAction: onNext,
            showBackButton: false
        ) {
            VStack {
                Text("Welcome to")
                    .font(.system(size: 32))
                    .foregroundColor(DVDesignSystem.Colors.text)
                    .fontWeight(.bold)
                
                Text("Design Village!")
                    .font(.system(size: 40, weight: .black))
                    .tracking(1)
                    .foregroundStyle(
                        LinearGradient(
                            colors: [DVDesignSystem.Colors.text, DVDesignSystem.Colors.textSecondary],
                            startPoint: .top,
                            endPoint: .bottom
                        )
                    )
                    .padding(.bottom, 15)
                
                Image("DVLogo")
                    .resizable()
                    .scaledToFit()
                    .frame(width: 250, height: 150)
                    .cornerRadius(20)
                    .padding(.bottom, 0)
                    .shadow(color: DVDesignSystem.Colors.shadowColor, radius: 15)
                
                VStack(spacing: 15) {
                    Text("NEXUS")
                        .font(.system(size: 34, weight: .black))
                        .tracking(8)
                        .foregroundColor(DVDesignSystem.Colors.text)
                    
                    DVDesignSystem.Effects.accentLine()
                        .frame(width: 200)
                }
                .padding(.bottom, 30)
                
                Text("Let's get you oriented.")
                    .font(.system(size: 24, weight: .semibold))
                    .foregroundColor(DVDesignSystem.Colors.textSecondary)
                    .multilineTextAlignment(.center)
                    .padding(.horizontal, 30)
            }
        }
    }
}

struct DVOverviewSlide: View {
    let onNext: () -> Void
    let onBack: () -> Void
    
    var body: some View {
        DVBaseSlide(
            buttonText: "Next",
            buttonAction: onNext,
            onBack: onBack
        ) {
            VStack {
                DVTitleWithShadow(
                    text: "The DV app has:",
                    font: .system(size: 38, weight: .bold)
                )
                .padding(.bottom, 50)
                
                VStack(alignment: .leading, spacing: 30) {
                    DVIconFeatureRow(icon: "map.fill", text: "A Map of the area")
                    DVIconFeatureRow(icon: "calendar", text: "Schedule for events")
                    DVIconFeatureRow(icon: "info.circle.fill", text: "Info on the history")
                }
                .padding(.horizontal)
            }
        }
    }
}

struct DVRulesSlide: View {
    @Binding var selectedRole: DVRole
    let onNext: () -> Void
    let onBack: () -> Void
    
    @State private var hasSelectedRole = false
    
    private func competitorRules() -> [(emoji: String, text: String)] {
        [
            (emoji: "🏠", text: "Build a stable shelter"),
            (emoji: "🌿", text: "Preserve the site"),
            (emoji: "📝", text: "Submit documentation"),
            (emoji: "🌙", text: "Stay overnight"),
            (emoji: "⚠️", text: "Follow safety rules")
        ]
    }
    
    private func visitorRules() -> [(emoji: String, text: String)] {
        [
            (emoji: "👥", text: "Follow staff guidance"),
            (emoji: "🚶‍♂️", text: "Stay in visitor areas"),
            (emoji: "🏗️", text: "Respect competitor spaces"),
            (emoji: "✨", text: "Keep a safe distance")
        ]
    }
    
    private func welcomeMessage(for role: DVRole) -> String {
        switch role {
        case .competitor:
            return "Time to build something amazing!"
        case .visitor:
            return "Get ready to be inspired!"
        }
    }
    
    var body: some View {
        DVBaseSlide(
            buttonText: "Next",
            buttonDisabled: !hasSelectedRole,
            buttonAction: onNext,
            onBack: onBack
        ) {
            VStack(spacing: 30) {
                if !hasSelectedRole {
                    DVTitleWithShadow(
                        text: "How are you experiencing Design Village?",
                        font: .system(size: 32, weight: .bold)
                    )
                    .multilineTextAlignment(.center)
                    .padding(.bottom, 20)
                    
                    HStack(spacing: 20) {
                        DVRoleButton(
                            title: "Competitor",
                            emoji: "🏃‍♂️",
                            isSelected: selectedRole == .competitor
                        ) {
                            withAnimation {
                                selectedRole = .competitor
                                hasSelectedRole = true
                            }
                        }
                        
                        DVRoleButton(
                            title: "Visitor",
                            emoji: "👥",
                            isSelected: selectedRole == .visitor
                        ) {
                            withAnimation {
                                selectedRole = .visitor
                                hasSelectedRole = true
                            }
                        }
                    }
                    .padding(.horizontal)
                } else {
                    VStack(spacing: 16) {
                        DVTitleWithShadow(
                            text: selectedRole == .competitor ? "Competitor Rules" : "Visitor Guidelines",
                            font: .system(size: 38, weight: .bold)
                        )
                        
                        Text(welcomeMessage(for: selectedRole))
                            .font(.system(size: 20, weight: .medium))
                            .foregroundColor(DVDesignSystem.Colors.textSecondary)
                            .multilineTextAlignment(.center)
                            .padding(.horizontal)
                            .padding(.bottom, 20)
                    }
                    
                    if selectedRole == .competitor {
                        VStack(spacing: 16) {
                            ForEach(competitorRules(), id: \.text) { rule in
                                OnboardingCompactRuleRow(emoji: rule.emoji, text: rule.text)
                            }
                            
                            OnboardingCallout(text: "Visit Rules tab for complete guidelines.")
                                .padding(.top, 20)
                        }
                    } else {
                        VStack(spacing: 24) {
                            ForEach(visitorRules(), id: \.text) { rule in
                                DVFeatureRow(emoji: rule.emoji, text: rule.text)
                            }
                        }
                    }
                }
            }
            .padding(.horizontal)
        }
    }
}

// MARK: - Onboarding-specific Components

private struct OnboardingCompactRuleRow: View {
    let emoji: String
    let text: String
    
    var body: some View {
        HStack(spacing: 12) {
            Text(emoji)
                .font(.system(size: 24))
                .frame(width: 32)
            
            Text(text)
                .font(.system(size: 18, weight: .medium))
                .foregroundColor(DVDesignSystem.Colors.text)
            
            Spacer()
        }
        .padding(.vertical, 10)
        .padding(.horizontal, 16)
        .background(
            RoundedRectangle(cornerRadius: 12)
                .fill(DVDesignSystem.Colors.surface)
                .shadow(color: DVDesignSystem.Colors.shadowColor, radius: 2, x: 0, y: 1)
        )
        .overlay(
            RoundedRectangle(cornerRadius: 12)
                .strokeBorder(DVDesignSystem.Colors.divider, lineWidth: 1)
        )
    }
}

private struct OnboardingCallout: View {
    let text: String
    
    var body: some View {
        Text(text)
            .font(.system(size: 16, weight: .medium))
            .foregroundColor(DVDesignSystem.Colors.text)
            .multilineTextAlignment(.center)
            .padding(16)
            .frame(maxWidth: .infinity)
            .background(
                RoundedRectangle(cornerRadius: 12)
                    .fill(DVDesignSystem.Colors.yellow.opacity(0.2))
            )
            .overlay(
                RoundedRectangle(cornerRadius: 12)
                    .strokeBorder(DVDesignSystem.Colors.yellow.opacity(0.5), lineWidth: 1)
            )
    }
}

struct DVPolyCanyonSlide: View {
    let onNext: () -> Void
    let onBack: () -> Void
    
    var body: some View {
        DVBaseSlide(
            buttonText: "Next",
            buttonAction: onNext,
            onBack: onBack
        ) {
            VStack {
                DVTitleWithShadow(
                    text: "Curious about\nthe canyon?",
                    font: .system(size: 38, weight: .bold)
                )
                .multilineTextAlignment(.center)
                .lineSpacing(2)
                .padding(.bottom, 10)
                
                Image("PCOverview")
                    .resizable()
                    .scaledToFit()
                    .frame(minHeight: 200)
                    .cornerRadius(20)
                    .padding(.bottom, 20)
                    .shadow(color: DVDesignSystem.Colors.shadowColor, radius: 8, x: 0, y: 4)
                
                DVSettingsPrompt(text: "Switch to 'Poly Canyon' to explore the structures!")
                    .padding(.horizontal)
            }
        }
    }
}

struct DVFinalSlide: View {
    let onComplete: () -> Void
    @State private var isAnimating = false
    @State private var shouldComplete = false
    @State private var isButtonPressed = false
    
    var body: some View {
        GeometryReader { geometry in
            ZStack {
                DVDesignSystem.Colors.background
                    .ignoresSafeArea()
                
                VStack(spacing: 30) {
                    Spacer()
                    
                    Image("DVNexus")
                        .resizable()
                        .scaledToFit()
                        .frame(height: 350)
                        .scaleEffect(shouldComplete ? 12 : 1)
                        .opacity(shouldComplete ? 0 : 1)
                    
                    VStack(spacing: 0) {
                        DVTitleWithShadow(
                            text: "It takes a village.",
                            font: .system(size: 36, weight: .bold)
                        )
                        .padding(.top, 15)
                        
                        Text("Welcome to the nexus\n of creative connections!")
                            .font(.system(size: 24, weight: .medium))
                            .foregroundColor(DVDesignSystem.Colors.textSecondary)
                            .multilineTextAlignment(.center)
                            .padding(.top, 5)
                    }
                    .opacity(shouldComplete ? 0 : 1)
                    
                    Spacer()
                    
                    Button {
                        withAnimation(.easeInOut(duration: 0.8)) {
                            shouldComplete = true
                        }
                        
                        DispatchQueue.main.asyncAfter(deadline: .now() + 0.7) {
                            onComplete()
                        }
                    } label: {
                        Text("Enter Design Village")
                            .font(.system(size: 22, weight: .bold))
                            .foregroundColor(DVDesignSystem.Colors.text)
                            .frame(width: 250)
                            .padding(.vertical, 16)
                            .background(
                                ZStack {
                                    RoundedRectangle(cornerRadius: 30)
                                        .fill(DVDesignSystem.Colors.yellow)
                                    
                                    // Pulsing animation
                                    if !shouldComplete {
                                        RoundedRectangle(cornerRadius: 30)
                                            .stroke(
                                                LinearGradient(
                                                    colors: [
                                                        DVDesignSystem.Colors.orange,
                                                        DVDesignSystem.Colors.teal
                                                    ],
                                                    startPoint: .leading,
                                                    endPoint: .trailing
                                                ),
                                                lineWidth: 3
                                            )
                                            .scaleEffect(isAnimating ? 1.1 : 1.0)
                                            .opacity(isAnimating ? 0.5 : 1.0)
                                    }
                                }
                            )
                            .scaleEffect(isButtonPressed ? 0.96 : 1.0)
                            .shadow(color: DVDesignSystem.Colors.shadowColor, radius: 4, x: 0, y: 2)
                    }
                    .pressAction {
                        withAnimation(.spring(response: 0.2, dampingFraction: 0.7)) {
                            isButtonPressed = true
                        }
                    } onRelease: {
                        withAnimation(.spring(response: 0.2, dampingFraction: 0.7)) {
                            isButtonPressed = false
                        }
                    }
                    .opacity(shouldComplete ? 0 : 1)
                    .padding(.bottom, 50)
                }
            }
        }
        .onAppear {
            withAnimation(
                .easeInOut(duration: 1.5)
                .repeatForever(autoreverses: true)
            ) {
                isAnimating = true
            }
        }
    }
}

struct DVOnboarding_Previews: PreviewProvider {
    static var previews: some View {
        DVOnboarding(userRole: .constant(.visitor))
    }
}

File: react/src/routing/DesignVillage/DVOnboarding.js
import React, { useState, useRef } from "react";
import {
  View,
  Text,
  Image,
  TouchableOpacity,
  StyleSheet,
  Animated,
} from "react-native";
import AsyncStorage from "@react-native-async-storage/async-storage";

// Import images (using the images for slide 1 and 4)
// Make sure these paths are correct in your project.
import DVLogo from "./Images/DVLogo.png";
import OGDefault from "./Images/OGDefault.png";
// import DVNexus from './Images/DVNexus.png'; // Final slide image is commented out per instructions

// A simple role "enum"
const DVRole = {
  VISITOR: "visitor",
  COMPETITOR: "competitor",
};

const DVOnboarding = ({ onComplete }) => {
  const [currentSlide, setCurrentSlide] = useState(0);
  const [selectedRole, setSelectedRole] = useState(DVRole.VISITOR);
  const [onboardingComplete, setOnboardingComplete] = useState(false);

  const totalSlides = 5;

  const goToNextSlide = () => {
    if (currentSlide < totalSlides - 1) {
      setCurrentSlide(currentSlide + 1);
    }
  };

  const goToPreviousSlide = () => {
    if (currentSlide > 0) {
      setCurrentSlide(currentSlide - 1);
    }
  };

  const completeOnboarding = async () => {
    try {
      await AsyncStorage.setItem("DVUserRole", selectedRole);
      setOnboardingComplete(true);
      onComplete(selectedRole);
    } catch (error) {
      console.error("Error saving user role:", error);
    }
  };

  if (onboardingComplete) {
    return null;
  }

  // Render one slide at a time based on currentSlide
  let slideComponent;
  switch (currentSlide) {
    case 0:
      slideComponent = <DVWelcomeSlide onNext={goToNextSlide} />;
      break;
    case 1:
      slideComponent = (
        <DVOverviewSlide onNext={goToNextSlide} onBack={goToPreviousSlide} />
      );
      break;
    case 2:
      slideComponent = (
        <DVRulesSlide
          selectedRole={selectedRole}
          setSelectedRole={setSelectedRole}
          onNext={goToNextSlide}
          onBack={goToPreviousSlide}
        />
      );
      break;
    case 3:
      slideComponent = (
        <DVPolyCanyonSlide onNext={goToNextSlide} onBack={goToPreviousSlide} />
      );
      break;
    case 4:
      slideComponent = <DVFinalSlide onComplete={completeOnboarding} />;
      break;
    default:
      slideComponent = null;
  }

  return (
    <View style={styles.container}>
      {slideComponent}
      {currentSlide < 4 && (
        <DVOnboardingIndicator
          totalStages={totalSlides}
          currentStage={currentSlide}
        />
      )}
    </View>
  );
};

// A base slide component that provides layout and navigation buttons.
const DVBaseSlide = ({
  children,
  buttonText = "Next",
  buttonAction,
  buttonDisabled = false,
  showBackButton = true,
  onBack,
}) => {
  return (
    <View style={styles.baseSlide}>
      <View style={styles.contentContainer}>{children}</View>
      <View style={styles.navigationContainer}>
        <TouchableOpacity
          style={[styles.navButton, buttonDisabled && styles.navButtonDisabled]}
          onPress={buttonAction}
          disabled={buttonDisabled}
        >
          <Text style={styles.navButtonText}>{buttonText}</Text>
        </TouchableOpacity>
        {showBackButton && onBack && (
          <TouchableOpacity onPress={onBack}>
            <Text style={styles.backButton}>Back</Text>
          </TouchableOpacity>
        )}
      </View>
    </View>
  );
};

// Slide 1: Welcome
const DVWelcomeSlide = ({ onNext }) => {
  return (
    <DVBaseSlide buttonText="Next" buttonAction={onNext} showBackButton={false}>
      <View style={styles.centerContent}>
        <Text style={styles.welcomeTitle}>Welcome to</Text>
        <Text style={styles.welcomeSubtitle}>Design Village!</Text>
        <Image source={DVLogo} style={styles.logoImage} resizeMode="contain" />
        <Text style={styles.orientationText}>Let's get you oriented.</Text>
      </View>
    </DVBaseSlide>
  );
};

// Slide 2: Overview
const DVOverviewSlide = ({ onNext, onBack }) => {
  return (
    <DVBaseSlide buttonText="Next" buttonAction={onNext} onBack={onBack}>
      <View style={styles.centerContent}>
        <Text style={styles.overviewTitle}>The DV app has:</Text>
        <View style={styles.featureList}>
          <DVFeatureRow icon="map" text="A Map of the area" />
          <DVFeatureRow icon="calendar" text="Schedule for events" />
          <DVFeatureRow icon="info-circle" text="Info on the history" />
        </View>
      </View>
    </DVBaseSlide>
  );
};

// A callout component for important messages
const DVCallout = ({ text }) => {
  return (
    <View style={styles.calloutContainer}>
      <Text style={styles.calloutIcon}>📋</Text>
      <Text style={styles.calloutText}>{text}</Text>
    </View>
  );
};

// A compact rule row specifically for competitor rules
const DVCompactRuleRow = ({ icon, text }) => {
  return (
    <View style={styles.compactRuleRow}>
      <Text style={styles.compactRuleIcon}>{icon}</Text>
      <Text style={styles.compactRuleText}>{text}</Text>
    </View>
  );
};

// Slide 3: Rules (with a two-stage experience)
const DVRulesSlide = ({ selectedRole, setSelectedRole, onNext, onBack }) => {
  const [hasSelectedRole, setHasSelectedRole] = useState(false);

  const handleSelectRole = (role) => {
    setSelectedRole(role);
    setHasSelectedRole(true);
  };

  // Helper function to get the correct rules based on role
  const getRoleRules = (role) => {
    if (role === DVRole.COMPETITOR) {
      return [
        { icon: "🏠", text: "Build a stable shelter" },
        { icon: "🌿", text: "Preserve the site" },
        { icon: "📝", text: "Submit documentation" },
        { icon: "🌙", text: "Stay overnight" },
        { icon: "⚠️", text: "Follow safety rules" },
      ];
    } else {
      return [
        { icon: "👥", text: "Follow event staff guidance" },
        { icon: "🚶‍♂️", text: "Stay in visitor areas" },
        { icon: "🏗️", text: "Respect competitor spaces" },
        { icon: "✨", text: "Keep a safe distance" },
      ];
    }
  };

  return (
    <DVBaseSlide
      buttonText="Next"
      buttonAction={onNext}
      onBack={onBack}
      buttonDisabled={!hasSelectedRole}
    >
      <View style={styles.centerContent}>
        {!hasSelectedRole ? (
          <>
            <Text style={styles.rulesQuestion}>
              How are you experiencing Design Village?
            </Text>
            <View style={styles.rolesContainer}>
              <DVRoleButton
                title="Competitor"
                icon="🏃‍♂️"
                isSelected={selectedRole === DVRole.COMPETITOR}
                onPress={() => handleSelectRole(DVRole.COMPETITOR)}
              />
              <DVRoleButton
                title="Visitor"
                icon="👥"
                isSelected={selectedRole === DVRole.VISITOR}
                onPress={() => handleSelectRole(DVRole.VISITOR)}
              />
            </View>
          </>
        ) : (
          <>
            <Text style={styles.rulesSelected}>
              {selectedRole === DVRole.COMPETITOR
                ? "Competitor Rules"
                : "Visitor Guidelines"}
            </Text>
            <View style={styles.rulesContainer}>
              {selectedRole === DVRole.COMPETITOR ? (
                <>
                  <View style={styles.compactRulesList}>
                    {getRoleRules(selectedRole).map((rule, index) => (
                      <DVCompactRuleRow
                        key={index}
                        icon={rule.icon}
                        text={rule.text}
                      />
                    ))}
                  </View>
                  <DVCallout text="Visit the Rules tab for complete competition guidelines. All rules must be followed to participate." />
                </>
              ) : (
                <View style={styles.featureList}>
                  {getRoleRules(selectedRole).map((rule, index) => (
                    <DVFeatureRow
                      key={index}
                      icon={rule.icon}
                      text={rule.text}
                    />
                  ))}
                </View>
              )}
            </View>
          </>
        )}
      </View>
    </DVBaseSlide>
  );
};

// Slide 4: Poly Canyon Info
const DVPolyCanyonSlide = ({ onNext, onBack }) => {
  return (
    <DVBaseSlide buttonText="Next" buttonAction={onNext} onBack={onBack}>
      <View style={styles.centerContent}>
        <Text style={styles.polyTitle}>Curious about{"\n"}the canyon?</Text>
        <Image source={OGDefault} style={styles.polyImage} resizeMode="cover" />
        <DVSettingsPrompt text="Switch to 'Poly Canyon' to explore the structures!" />
      </View>
    </DVBaseSlide>
  );
};

// Slide 5: Final slide (image commented out)
const DVFinalSlide = ({ onComplete }) => {
  const [shouldComplete, setShouldComplete] = useState(false);
  const scaleAnim = useRef(new Animated.Value(1)).current;

  const handleEnter = () => {
    // Animate a scaling effect similar to the SwiftUI animation
    Animated.timing(scaleAnim, {
      toValue: 12,
      duration: 800,
      useNativeDriver: true,
    }).start(() => {
      setShouldComplete(true);
      setTimeout(() => {
        onComplete();
      }, 700);
    });
  };

  return (
    <View style={styles.finalContainer}>
      <View style={styles.finalContent}>
        {/*
        <Animated.Image
          source={DVNexus}
          style={[
            styles.finalImage,
            { transform: [{ scale: scaleAnim }], opacity: shouldComplete ? 0 : 1 },
          ]}
          resizeMode="contain"
        />
        */}
        <Text style={styles.finalTitle}>It takes a village.</Text>
        <Text style={styles.finalSubtitle}>
          Welcome to the nexus of creative connections!
        </Text>
      </View>
      <TouchableOpacity
        style={styles.enterButton}
        onPress={handleEnter}
        disabled={shouldComplete}
      >
        <Text style={styles.enterButtonText}>Enter Design Village</Text>
      </TouchableOpacity>
    </View>
  );
};

// An indicator at the top showing the current slide (hidden on the final slide)
const DVOnboardingIndicator = ({ totalStages, currentStage }) => {
  return (
    <View style={styles.indicatorContainer}>
      {Array.from({ length: totalStages }).map((_, index) => (
        <View
          key={index}
          style={[
            styles.indicatorDot,
            index === currentStage
              ? styles.indicatorDotActive
              : styles.indicatorDotInactive,
          ]}
        />
      ))}
    </View>
  );
};

// A row to display a feature with an icon and text.
const DVFeatureRow = ({ icon, text }) => {
  // Map text descriptions to actual icons
  const getIcon = (iconName) => {
    switch (iconName) {
      case "map":
        return "🗺️";
      case "calendar":
        return "📅";
      case "info-circle":
        return "ℹ️";
      default:
        return iconName; // For cases where we already pass an emoji
    }
  };

  return (
    <View style={styles.featureRow}>
      <Text style={styles.featureIcon}>{getIcon(icon)}</Text>
      <Text style={styles.featureText}>{text}</Text>
    </View>
  );
};

// A button used in the Rules slide for selecting a role.
const DVRoleButton = ({ title, icon, isSelected, onPress }) => {
  return (
    <TouchableOpacity
      style={[
        styles.roleButton,
        isSelected ? styles.roleButtonSelected : styles.roleButtonUnselected,
      ]}
      onPress={onPress}
    >
      <Text style={[styles.roleIcon, isSelected && styles.roleIconSelected]}>
        {icon}
      </Text>
      <Text style={[styles.roleTitle, isSelected && styles.roleTitleSelected]}>
        {title}
      </Text>
    </TouchableOpacity>
  );
};

// A prompt with a gear icon and text.
const DVSettingsPrompt = ({ text }) => {
  return (
    <View style={styles.settingsPrompt}>
      <Text style={styles.settingsIcon}>⚙️</Text>
      <Text style={styles.settingsText}>{text}</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#fff",
  },
  completeContainer: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
  },
  completeText: {
    fontSize: 24,
    fontWeight: "bold",
  },
  baseSlide: {
    flex: 1,
    justifyContent: "space-between",
    paddingHorizontal: 20,
    paddingTop: 100,
    paddingBottom: 60,
  },
  contentContainer: {
    flex: 1,
    alignItems: "center",
  },
  navigationContainer: {
    alignItems: "center",
  },
  navButton: {
    backgroundColor: "black",
    paddingHorizontal: 20,
    paddingVertical: 12,
    borderRadius: 25,
    marginBottom: 10,
  },
  navButtonDisabled: {
    backgroundColor: "gray",
  },
  navButtonText: {
    color: "white",
    fontSize: 18,
    fontWeight: "bold",
  },
  backButton: {
    fontSize: 16,
    fontWeight: "500",
    color: "black",
    textDecorationLine: "underline",
  },
  centerContent: {
    alignItems: "center",
  },
  welcomeTitle: {
    fontSize: 32,
    fontWeight: "bold",
    color: "black",
    marginBottom: 0,
  },
  welcomeSubtitle: {
    fontSize: 50,
    fontWeight: "bold",
    color: "black",
    marginBottom: 15,
  },
  logoImage: {
    width: 200,
    height: 200,
    borderRadius: 40,
    marginBottom: 30,
  },
  orientationText: {
    fontSize: 28,
    fontWeight: "600",
    color: "rgba(0,0,0,0.8)",
    textAlign: "center",
    paddingHorizontal: 30,
  },
  overviewTitle: {
    fontSize: 38,
    fontWeight: "bold",
    color: "black",
    marginBottom: 50,
  },
  featureList: {
    width: "100%",
    alignItems: "center",
    paddingHorizontal: 20,
  },
  featureRow: {
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "flex-start",
    marginVertical: 12,
    width: "100%",
    paddingHorizontal: 30,
  },
  featureIcon: {
    fontSize: 34,
    width: 45,
    textAlign: "center",
    marginRight: 15,
  },
  featureText: {
    flex: 1,
    fontSize: 24,
    fontWeight: "600",
    color: "black",
  },
  rulesQuestion: {
    fontSize: 32,
    fontWeight: "bold",
    color: "black",
    textAlign: "center",
    marginBottom: 40,
  },
  rolesContainer: {
    flexDirection: "row",
    justifyContent: "space-between",
    width: "100%",
    paddingHorizontal: 20,
  },
  roleButton: {
    flex: 1,
    height: 160,
    marginHorizontal: 10,
    borderRadius: 20,
    justifyContent: "center",
    alignItems: "center",
  },
  roleButtonSelected: {
    backgroundColor: "black",
  },
  roleButtonUnselected: {
    backgroundColor: "rgba(128,128,128,0.2)",
  },
  roleIcon: {
    fontSize: 40,
    fontWeight: "bold",
    marginBottom: 10,
    color: "black",
  },
  roleIconSelected: {
    color: "white",
  },
  roleTitle: {
    fontSize: 22,
    fontWeight: "bold",
    color: "black",
  },
  roleTitleSelected: {
    color: "white",
  },
  rulesSelected: {
    fontSize: 32,
    fontWeight: "bold",
    color: "black",
    marginBottom: 25,
    textAlign: "center",
  },
  rulesContainer: {
    width: "100%",
    alignItems: "center",
    paddingHorizontal: 20,
  },
  compactRulesList: {
    width: "100%",
    marginBottom: 20,
  },
  compactRuleRow: {
    flexDirection: "row",
    alignItems: "center",
    marginVertical: 8,
    width: "100%",
  },
  compactRuleIcon: {
    fontSize: 24,
    width: 40,
    textAlign: "center",
  },
  compactRuleText: {
    flex: 1,
    fontSize: 18,
    fontWeight: "600",
    color: "black",
    marginLeft: 12,
  },
  polyTitle: {
    fontSize: 38,
    fontWeight: "bold",
    color: "black",
    textAlign: "center",
    marginBottom: 10,
  },
  polyImage: {
    width: 330,
    height: 220,
    borderRadius: 25,
    marginBottom: 20,
  },
  settingsPrompt: {
    flexDirection: "row",
    alignItems: "center",
    backgroundColor: "rgba(128,128,128,0.15)",
    paddingHorizontal: 20,
    paddingVertical: 16,
    borderRadius: 16,
    marginHorizontal: 40,
  },
  settingsIcon: {
    fontSize: 24,
    fontWeight: "bold",
    marginRight: 12,
  },
  settingsText: {
    fontSize: 20,
    fontWeight: "500",
  },
  finalContainer: {
    flex: 1,
    justifyContent: "space-between",
    paddingHorizontal: 20,
    paddingVertical: 60,
    alignItems: "center",
  },
  finalContent: {
    alignItems: "center",
  },
  finalImage: {
    width: 200,
    height: 200,
    marginBottom: 30,
  },
  finalTitle: {
    fontSize: 36,
    fontWeight: "bold",
    color: "black",
    textAlign: "center",
    marginBottom: 15,
  },
  finalSubtitle: {
    fontSize: 24,
    fontWeight: "500",
    color: "black",
    textAlign: "center",
  },
  enterButton: {
    backgroundColor: "black",
    paddingHorizontal: 20,
    paddingVertical: 16,
    borderRadius: 30,
    width: 250,
    alignItems: "center",
    marginBottom: 50,
  },
  enterButtonText: {
    fontSize: 22,
    fontWeight: "bold",
    color: "white",
  },
  indicatorContainer: {
    position: "absolute",
    top: 20,
    alignSelf: "center",
    flexDirection: "row",
    backgroundColor: "rgba(128,128,128,0.1)",
    paddingHorizontal: 20,
    paddingVertical: 8,
    borderRadius: 20,
    shadowColor: "#000",
    shadowOpacity: 0.1,
    shadowRadius: 4,
    shadowOffset: { width: 0, height: 2 },
  },
  indicatorDot: {
    marginHorizontal: 6,
  },
  indicatorDotActive: {
    width: 12,
    height: 12,
    borderRadius: 6,
    backgroundColor: "black",
  },
  indicatorDotInactive: {
    width: 8,
    height: 8,
    borderRadius: 4,
    backgroundColor: "rgba(128,128,128,0.5)",
  },
  calloutContainer: {
    flexDirection: "row",
    alignItems: "center",
    backgroundColor: "rgba(0,0,0,0.05)",
    padding: 16,
    borderRadius: 16,
    width: "100%",
    borderWidth: 1,
    borderColor: "rgba(0,0,0,0.1)",
  },
  calloutIcon: {
    fontSize: 24,
    marginRight: 12,
  },
  calloutText: {
    flex: 1,
    fontSize: 18,
    fontWeight: "600",
    color: "black",
    lineHeight: 24,
  },
});

export default DVOnboarding;
